# Drill - Exercises in Bayes' Rule

** Unit 3 / Lesson 2 / Project 6 **

A diagnostic test has a 98% probability of giving a positive result when applied to a person suffering from Thripshaw's Disease, and 10% probability of giving a (false) positive when applied to a non-sufferer. It is estimated that 0.5 % of the population are sufferers. Suppose that the test is now administered to a person whose disease status is unknown. Calculate the probability that the test will:

 - Be positive
 - Correctly diagnose a sufferer of Thripshaw's
 - Correctly identify a non-sufferer of Thripshaw's
 - Misclassify the person
 
First look over the numbers and estimate each of the four probabilities, using your intuition. 

Assume 1000 people
5 infected

      Actual condition
               True     False
    Pred  P    5*0.98   995*0.1
    Pred  F    5*0.02   995*0.9     


In [49]:
Pos = (0.1*0.995 + 0.98*0.005)
print("Be positive: "+ str(round(Pos,4)))

Be positive: 0.1044


In [51]:
CorT = 5*0.98/(5*0.98 + 5*0.02)*100
print("Correctly diagnose a sufferer of Thripshaw's: " + str(round(CorT,2)) + "%")

Correctly diagnose a sufferer of Thripshaw's: 98.0%


In [52]:
CorNT = 995*0.9/(995*0.1 + 995*0.9)*100
print("Correctly identify a non-sufferer of Thripshaw's: " + str(round(CorNT,2)) + "%")

Correctly identify a non-sufferer of Thripshaw's: 90.0%


In [22]:
Missed = (5*0.02+995*0.1)/(5*0.98 + 995*0.1 + 5*0.02 + 995*0.9)*100
print("Misclassify the person: " + str(round(Missed,2)) + "%")

Misclassify the person: 9.96%


## Using Bayes' Rule:

 - Be positive

P(Positive) = (0.1*0.995 + 0.98*0.005)
       = 0.1044

 - Correctly diagnose a sufferer of Thripshaw's

P(Positive | Thirp) = P(Thrip | Pos)* P(Pos)/P(Thrip) 
               
 - Correctly identify a non-sufferer of Thripshaw's

P(Neg | Not Thrip) = P(Not Thrip | Neg)* P(Neg) / P(Not Thrip)

Misclassify the person

In [40]:
Prob_Pos = (0.1*0.995 + 0.98*0.005)
Prob_Thrip = 0.005
Prob_Neg = (0.9*0.995 + 0.02*0.005)
Prob_Not_Thrip = 0.995
Prob_Pos_if_Thrip = (0.005*0.98/(0.005*0.98 + 0.995*0.1))*Prob_Pos / Prob_Thrip
Prob_Neg_ifnot_Thrip = (0.9*0.995/(0.995*0.9 + 0.02*0.005))*Prob_Neg / Prob_Not_Thrip
Prob_Misclassify = (5*0.02+995*0.1)/(5*0.98 + 995*0.1 + 5*0.02 + 995*0.9)

In [41]:
Prob_Pos_if_Thrip

0.98

In [42]:
Prob_Neg_ifnot_Thrip

0.8999999999999999

In [39]:
Prob_Misclassify

0.0996